### ISSUE TITLE
Call instantly declined

### ISSUE THREAD

**Issue by burruplambert** [pytgcalls/pytgcalls#250]
**Describe the bug**

My calls seem to be intermittent. They were working fine, but now I receive an instant 'declined call' notification in the chat for my test client. The test client does not ring.

**To Reproduce**
Run code
```
client = TelegramClient(
    'anon',
    api_id=TELEGRAM_API_ID,
    api_hash=TELEGRAM_API_HASH,
)

app = PyTgCalls(client)
app.start()
chat_id = xxxx
app.play(
    chat_id,
)
idle()
```

**Expected behavior**
The call to ring.

**Log**
```
PyTgCalls v2.0.4 powered by NTgCalls v1.2.3, Copyright (C) 2021-2024 Laky-64 <https://github.com/Laky-64>
Licensed under the terms of the GNU Lesser General Public License v3 or later (LGPLv3+)

Future exception was never retrieved
future: <Future finished exception=CallDiscarded('Call discarded by xxxx')>
pytgcalls.exceptions.CallDiscarded: Call discarded by xxxx
Traceback (most recent call last):
  File "/home/linuxuser/test/api/testing_telegram.py", line 57, in <module>
    app.play(
  File "/home/linuxuser/test/api/venv/lib/python3.12/site-packages/pytgcalls/sync.py", line 55, in async_to_sync_wrap
    return loop.run_until_complete(coroutine)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/linuxuser/test/api/venv/lib/python3.12/site-packages/pytgcalls/mutex.py", line 9, in async_wrapper
    return await func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/linuxuser/test/api/venv/lib/python3.12/site-packages/pytgcalls/statictypes.py", line 96, in async_wrapper
    return await func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/linuxuser/test/api/venv/lib/python3.12/site-packages/pytgcalls/mtproto_required.py", line 23, in async_wrapper
    return await func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/linuxuser/test/api/venv/lib/python3.12/site-packages/pytgcalls/methods/stream/play.py", line 123, in play
    result: RawCallUpdate = await asyncio.wait_for(
                            ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/asyncio/tasks.py", line 520, in wait_for
    return await fut
           ^^^^^^^^^
pytgcalls.exceptions.CallDeclined: Call declined by xxxx
```

**Linux (please complete the following information):**
- Python 3.12.4
- Ubuntu 22.04

**Additional context**
Was working fine.

**Comment by KSKOP69** [comment]
from telethon import TelegramClient
from pytgcalls import PyTgCalls
from pytgcalls import idle
from pytgcalls.types import MediaStream

api_id = 
api_hash = ""

client = TelegramClient("testing", api_id, api_hash)

app = PyTgCalls(client)
app.start()
chat_id = -1001234567890
app.play(
    chat_id,
    MediaStream(
                        'https://telegra.ph/file/de3464aa7d6bfafdd2dc3.mp4',
                    )
                )

idle()

**Comment by Laky-64** [comment]
> ```
> from telethon import TelegramClient
> from pytgcalls import PyTgCalls
> from pytgcalls import idle
> from pytgcalls.types import MediaStream
> 
> api_id = 
> api_hash = ""
> 
> client = TelegramClient("testing", api_id, api_hash)
> 
> app = PyTgCalls(client)
> app.start()
> chat_id = -1001234567890
> app.play(
>     chat_id,
>     MediaStream(
>                         'https://telegra.ph/file/de3464aa7d6bfafdd2dc3.mp4',
>                     )
>                 )
> 
> idle()
> ```

How did you uploaded to telegraph? Since it is disabled

**Comment by KSKOP69** [comment]
@Laky-64 it is very old

**Comment by Laky-64** [comment]
> **Describe the bug**
> 
> My calls seem to be intermittent. They were working fine, but now I receive an instant 'declined call' notification in the chat for my test client. The test client does not ring.
> 
> **To Reproduce** Run code
> 
> ```
> client = TelegramClient(
>     'anon',
>     api_id=TELEGRAM_API_ID,
>     api_hash=TELEGRAM_API_HASH,
> )
> 
> app = PyTgCalls(client)
> app.start()
> chat_id = xxxx
> app.play(
>     chat_id,
> )
> idle()
> ```
> 
> **Expected behavior** The call to ring.
> 
> **Log**
> 
> ```
> PyTgCalls v2.0.4 powered by NTgCalls v1.2.3, Copyright (C) 2021-2024 Laky-64 <https://github.com/Laky-64>
> Licensed under the terms of the GNU Lesser General Public License v3 or later (LGPLv3+)
> 
> Future exception was never retrieved
> future: <Future finished exception=CallDiscarded('Call discarded by xxxx')>
> pytgcalls.exceptions.CallDiscarded: Call discarded by xxxx
> Traceback (most recent call last):
>   File "/home/linuxuser/test/api/testing_telegram.py", line 57, in <module>
>     app.play(
>   File "/home/linuxuser/test/api/venv/lib/python3.12/site-packages/pytgcalls/sync.py", line 55, in async_to_sync_wrap
>     return loop.run_until_complete(coroutine)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
>     return future.result()
>            ^^^^^^^^^^^^^^^
>   File "/home/linuxuser/test/api/venv/lib/python3.12/site-packages/pytgcalls/mutex.py", line 9, in async_wrapper
>     return await func(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/linuxuser/test/api/venv/lib/python3.12/site-packages/pytgcalls/statictypes.py", line 96, in async_wrapper
>     return await func(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/linuxuser/test/api/venv/lib/python3.12/site-packages/pytgcalls/mtproto_required.py", line 23, in async_wrapper
>     return await func(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/linuxuser/test/api/venv/lib/python3.12/site-packages/pytgcalls/methods/stream/play.py", line 123, in play
>     result: RawCallUpdate = await asyncio.wait_for(
>                             ^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.12/asyncio/tasks.py", line 520, in wait_for
>     return await fut
>            ^^^^^^^^^
> pytgcalls.exceptions.CallDeclined: Call declined by xxxx
> ```
> 
> **Linux (please complete the following information):**
> 
> * Python 3.12.4
> * Ubuntu 22.04
> 
> **Additional context** Was working fine.

That errors means that you're trying to call to an account that is in the same telegram client of the caller

**Comment by burruplambert** [comment]
> > ```
> > from telethon import TelegramClient
> > from pytgcalls import PyTgCalls
> > from pytgcalls import idle
> > from pytgcalls.types import MediaStream
> > 
> > api_id = 
> > api_hash = ""
> > 
> > client = TelegramClient("testing", api_id, api_hash)
> > 
> > app = PyTgCalls(client)
> > app.start()
> > chat_id = -1001234567890
> > app.play(
> >     chat_id,
> >     MediaStream(
> >                         'https://telegra.ph/file/de3464aa7d6bfafdd2dc3.mp4',
> >                     )
> >                 )
> > 
> > idle()
> > ```
> 
> How did you uploaded to telegraph? Since it is disabled

This may explain the inconsistent behavior.

On my phone I have 2 Telegram apps installed, 1 logged into the caller account, and 1 logged into the receiving account. I am also logged into the receiving account on desktop telegram.

My end goal is to be able to call myself, however I believe I need multiple accounts to do this

Sometimes I receive a straight declined call msg. Other times the desktop client will ring for maybe 0.5 seconds, then stop, and then I receive the declined call msg.

The behavior is very inconsistent. I started to receive

`ValueError: Could not find the input entity for PeerUser(user_id=`

So I created a new session. Same problem. I was using the user_id of the user. To resolve that I manually had to call client.get_entity using the username, if I tried to use the user_id I would get the ValueError. Then it began to work again, but now it has gone back to declined message straight away. I even created a new session and that one goes straight to declined msg.

I am thinking that maybe it has to do with the privacy settings for each account?

Each contact is in the others contact list. Both contacts have Phone Number privacy set to My Contacts.

Are there any specific settings I need on/or off for this to work? Or is there a simpler way to call myself?
